R/implicit_assignment_linter.R

```diff
@@ -116,6 +130,9 @@ implicit_assignment_linter <- function(except = c("bquote", "expression", "expr"
     bad_expr <- xml_find_all(xml, xpath)
 
     print_only <- !is.na(xml_find_first(bad_expr, "parent::expr[parent::exprlist and *[1][self::OP-LEFT-PAREN]]"))
+    if (allow_print) {
+      bad_expr <- bad_expr[!print_only]
```

Comment:
This works, I think it will be ideal to work this in to the up-front XPath logic, rather than including those hits in the search only to drop them.

Have a try and if you get stuck we can mark it as a follow-up issue.